frontend/src/containers/App.js

```diff
@@ -0,0 +1,194 @@
+          // this in the eventual FE component.
+          let tracks = egressMediaStream.getTracks()
+          for (var i = 0; i < tracks.length; i++) {
+            pc.addTrack(tracks[i], egressMediaStream)
+          }
+        }
+
+        // Store this peer connection in RoomyPeerConnection map
+        roomyPcs[newPeerId] = pc;
+
+        // Setup handlers for when we receive data back on this peer connection.
+        pc.ontrack = function(event) {
+          let fromUserId = newPeerId.split("-")[1];
+          if (event.streams.length > 0 && fromUserId >= 0) {
+            console.log('setting up media for userId=%o', fromUserId)
+            // TODO: Add to grid component somehow.
+            // Keep video refs for now?
+          }
+        }
+
+        // Setup handler to monitor ICE candidates we can use on the peer
+        // connection.
+        pc.onicecandidate = function(event) {
+          console.log('received ICE candidate from newPeerId=%o', newPeerId)
+        }
+
+        // Add an offer on the peer connection and after setting the local
+        // description of the peer connection, emit the 'ReceiveMediaFrom'
+        // event using the SDP.
+        pc.createOffer(
+          function(localDescription) {
+            console.log('set location description for newPeerId=%o', newPeerId)
+            pc.setLocalDescription(localDescription,
+            function() {
+              // With the local description, we can send the event. We will
+              // await a 'ReceiveMediaAnswer' event and set the remote
+              // description on this peer connection.
+              let data = {
+                'from_peer_id': newPeerId,
+                'to_peer_id':   myPeerId,
+                'desc':         btoa(JSON.stringify(pc.localDescription))
+              }
+              rmsClient.receiveMediaFrom(data, () => {
+                console.log('peerId=%o requested to receive media from peerId=%o', myPeerId, newPeerId);
+                rmsClient.awaitReceiveMediaAnswer((resp) => {
+                  console.log('received media answer resp=%o for peerId=%o', resp, newPeerId);
+                  // TODO: validate.
+                  let sdpAnswer = JSON.parse(atob(resp["sdp_answer"]));
+                  if (sdpAnswer !== '') {
+                    var remoteDescription = new RTCSessionDescription(sdpAnswer);
+                    var tmp = pc.setRemoteDescription(remoteDescription,
+                    function() {
+                      console.log('set remote description for peerId=%o', newPeerId);
+                    }, function (e) {
+                      console.log('error=%o setting remote description for peerId=%o', e, newPeerId);
+                    });
+                    console.log('remote description=%o for peerId=%o', newPeerId)
+                  }
+                });
+              });
+            });
+          }
+        );
+      });
+    });
```

Comment:
Seeing all of these brackets, parenthesis, and semicolons is death lol

Reply:
pretty nasty